validateApiDefinition.js

```diff
@@ -150,64 +151,125 @@ export const validateDefinition = async (apiDefinition, fileName, validationLeve
+    jarProcess.stdout.pipe(writeStream);
+    jarProcess.stderr.on('data', (data) => {
+      console.error(data.toString());
+    });
+
+    const javaClientOutput = await new Promise((resolve, reject) => {
+      jarProcess.on('close', () => {
+        const javaClientStdout = fs.readFileSync(tempFilePath, 'utf8'); // Read the contents of the temporary file
+
+        // Analyse java client output to check whether the provided API definition is accepted by API Manager
+        let isValid;
+        const regex = /Total Successful Files Count (\d+)/;
+        const match = javaClientStdout.match(regex);
+        if (match) {
+          const successfulFileCount = parseInt(match[1], 10);
+          successfulFileCount === 1 ? (isValid = true) : (isValid = false);
+          fs.unlinkSync(tempFilePath); // Delete the temporary file
+          resolve([isValid, javaClientStdout]);
+        } else {
+          console.log(`Failed to get the successful file count from the java client output`);
+          reject(new Error('Invalid Java client output'));
+        }
       });
+    });
+
+    const [isValid, javaClientStdout] = await javaClientOutput;
+
+    console.log("\n\u25A1 Validating " + fileName + " using validation level " + validationLevel + " ...\n");
+
+    // Iterate the results and select only those with severity of 0 (i.e. errors)
+    result = result.filter((r) => r.severity === 0);
+
+    // Replace the path field with a string representation of the path
+    result.forEach((r) => (r.path = r.path.join(".")));
+
+    // Remove code, severity and range fields from the result as those add no value to the output
+    result.forEach((r) => {
+      delete r.severity;
+      delete r.range;
+    });
+
+    improveErrorMessages(result);
+    
+    if (isValid) {
+      if (result.length > 0) {
+        console.log(
+          "\u2757 Validation passed with the below-listed errors, using may lead to functionality issues. API Manager 4.2.0 will " +
+          chalk.green("ACCEPT") + " this API definition.\n"
+        );
+      }
+
+      await logErrorOutput(result);
+
+      console.log(chalk.green.bold("\nValidation Passed\n"));
+    } else {
+      console.log("\u2757 Validation failed with the below-listed errors. API Manager 4.2.0 will " +
+      chalk.red("NOT ACCEPT") + " this API definition.\n");
+
+      console.log(chalk.blue("\nAPI Manager Backend validation results : \n"));
+      extractJavaClientOutput(javaClientStdout)
 
-      improveErrorMessages(result);
-      
       if (result.length > 0) {
+        console.log(chalk.blue("\nLinter validation results : \n"));
         await logErrorOutput(result);
-        console.log(chalk.red.bold("\nValidation Failed\n"));
-        return false;
-      } else {
-        console.log(chalk.green.bold("\nValidation Passed\n"));
-        return true;
       }
-    });
-  }
+
+      console.log(chalk.red.bold("\nValidation Failed\n"));
+    }
+    return isValid;
+  });
+}
+
 };
```

Comment:
Shall we fix the indentation issues in this file? Shall we run this file through a code formatter?

Reply:
done